src/ws/cockpitwebservice.c

```diff
@@ -1360,6 +1363,7 @@ cockpit_web_service_create_socket (const gchar **protocols,
  * @input_buffer: optional bytes already parsed after headers
  * @auth: authentication object
  * @creds: credentials of user or NULL for failed auth
+ * @for_tls_proxy: Accept (only) https:// origins on http connection by default
```

Comment:
What about "Assume that the Browser is making TLS connections that are terminated in a proxy."   Changing the default Origin is the effect, but there might be more.